### ISSUE TITLE
Watching of rust files not working in Vite

### ISSUE THREAD

**Issue by darichey** [wasm-tool/rollup-plugin-rust#41]
When using this plugin in Vite, modifying rust files does not trigger a reload. Here is a minimal repro: https://github.com/darichey/rollup-plugin-rust-watch-issue-demo

Using `vite --debug`, I can see this in the logs when modifying `lib.rs`:
```
vite:hmr [file change] src/lib.rs +5s
vite:hmr [no modules matched] src/lib.rs +0ms
```

It's possible that this is a Vite issue. It seems related to https://github.com/vitejs/vite/issues/7024.

**Comment by meesvermeulen** [comment]
I ran into the same issue.
As a workaround I added a `handleHotUpdate` hook to reload:
```js
let rustPluginInstance = rust();
rustPluginInstance["handleHotUpdate"] = (ctx) => {
  if (ctx.file.endsWith(".rs")) {
    console.log("Rust code changed");
    ctx.server.restart();
  }
};
```

**Comment by Pauan** [comment]
This is definitely a Vite bug. They claim that [they fixed it](https://github.com/vitejs/vite/pull/9723), but it seems that it's not fixed.

**Comment by Boscop** [comment]
@darichey Is this still an issue? I'm curious what's the current status.

And I'm curious what y'all are using nowadays.
Can you please recommend a build tool that I can use for building a Rust WASM (Leptos) frontend that needs to use NPM JS/TS libs (and with auto-watch-reloading)? 🙂
Trunk can't bundle those libs.

ChatGPT recommends Vite, but [`vite-plugin-rsw`](https://github.com/rwasm/vite-plugin-rsw) hasn't been updated in 3 years and has long-standing open issues.

And [`vite-plugin-wasm-pack`](https://github.com/nshen/vite-plugin-wasm-pack/issues/25) doesn't do auto-watch-reloading :/
